pkg/controller/stargazer.go

```diff
@@ -0,0 +1,247 @@
+	stopCh <-chan struct{}) *Controller {
+
+	var informer cache.SharedIndexInformer
+	var resType string
+
+	nimbessInformerFactory := unpinformer.NewSharedInformerFactory(kubeClient, time.Second*30)
+
+	if conf.Controllers.UNP {
+		informer = nimbessInformerFactory.Nimbess().V1().UnpConfigs().Informer()
+		resType = "unp"
+	}
+	c := newResourceController(kubeClient, eventHandler, informer, resType)
+
+	if err := c.Run(stopCh); err != nil {
+		log.Fatalf("Error running controller: %s, error: %v", resType, err)
+	}
+	return c
+}
+
+func newResourceController(client *nimbessclientset.Clientset, eventHandler handlers.Handler, informer cache.SharedIndexInformer, resourceType string) *Controller {
+	queue := workqueue.NewRateLimitingQueue(workqueue.DefaultControllerRateLimiter())
+	var newEvent Event
+	var err error
+	informer.AddEventHandler(cache.ResourceEventHandlerFuncs{
+		AddFunc: func(obj interface{}) {
+			newEvent.key, err = cache.MetaNamespaceKeyFunc(obj)
+			if err != nil {
+				utilruntime.HandleError(err)
+			}
+			newEvent.eventType = "create"
+			newEvent.resourceType = resourceType
+			log.WithField("pkg", "stargazer-"+resourceType).Infof("Processing add to %v: %s", resourceType, newEvent.key)
+			queue.Add(newEvent)
+		},
+		UpdateFunc: func(old, new interface{}) {
+			newEvent.key, err = cache.MetaNamespaceKeyFunc(old)
+			newEvent.eventType = "update"
+			newEvent.resourceType = resourceType
+			log.WithField("pkg", "stargazer-"+resourceType).Infof("Processing update to %v: %s", resourceType, newEvent.key)
+			if err == nil {
+				queue.Add(newEvent)
+			}
+		},
+		DeleteFunc: func(obj interface{}) {
+			newEvent.key, err = cache.DeletionHandlingMetaNamespaceKeyFunc(obj)
+			newEvent.eventType = "delete"
+			newEvent.resourceType = resourceType
+			newEvent.namespace = utils.GetObjectMetaData(obj).Namespace
+			log.WithField("pkg", "stargazer-"+resourceType).Infof("Processing delete to %v: %s", resourceType, newEvent.key)
+			if err == nil {
+				queue.Add(newEvent)
+			}
+		},
+	})
+
+	return &Controller{
+		logger:       log.WithField("pkg", "stargazer-"+resourceType),
+		clientset:    client,
+		informer:     informer,
+		queue:        queue,
+		eventHandler: eventHandler,
+	}
+}
+
+// Run starts the stargazer controller
+func (c *Controller) Run(stopCh <-chan struct{}) error {
+
+	c.logger.Info("Starting stargazer controller")
+	//serverStartTime = time.Now().Local()
+
+	go c.informer.Run(stopCh)
+
+	if !cache.WaitForCacheSync(stopCh, c.informer.HasSynced) {
+		utilruntime.HandleError(fmt.Errorf("timed out waiting for caches to sync"))
+		return fmt.Errorf("failed to wait for caches to sync")
+	}
+
+	c.logger.Info("Stargazer controller synced and ready")
+
+	go wait.Until(c.runWorker, time.Second, stopCh)
+	c.logger.Info("Stargazer controller started")
+	return nil
+}
+
+// HasSynced is required for the cache.Controller interface.
```

Comment:
guess you copied this from the google example. I saw that also but found I didn't need to implement these.